extension/js/common/core/pgp-key.ts

```diff
@@ -352,6 +352,10 @@ export class PgpKey {
    * This is used to figure out how recently was key updated, and if one key is newer than other.
    */
   public static lastSig = async (key: OpenPGP.key.Key): Promise<number> => {
+    // key is undefined only for X.509 keys
+    if (!key) {
+      return Date.now();
```

Comment:
todo - give this a look

Reply:
Make a key abstraction, and pass that around instead, that could be either pgp or smime key behind it

I'll make another issue